presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeIcebergGeneralQueries.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.testing.ExpectedQueryRunner;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import org.testng.annotations.Test;
+
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+
+public class TestPrestoNativeIcebergGeneralQueries
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createNativeIcebergQueryRunner(false, true);
+    }
+
+    @Override
+    protected ExpectedQueryRunner createExpectedQueryRunner()
+            throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createJavaIcebergQueryRunner(true);
+    }
+
+    @Override
+    protected void createTables()
+    {
+        createTableToTestHiddenColumns();
+    }
+
+    private void createTableToTestHiddenColumns()
+    {
+        QueryRunner javaQueryRunner = ((QueryRunner) getExpectedQueryRunner());
+        if (!javaQueryRunner.tableExists(getSession(), "test_hidden_columns")) {
+            javaQueryRunner.execute("CREATE TABLE test_hidden_columns AS SELECT * FROM tpch.tiny.region WHERE regionkey=0");
+            javaQueryRunner.execute("INSERT INTO test_hidden_columns SELECT * FROM tpch.tiny.region WHERE regionkey=1");
+        }
+    }
+
+    @Test
+    public void testPathHiddenColumn()
```

Comment:
Should we add these two test cases into `IcebergDistributedSmokeTestBase` or `IcebergDistributedTestBase` to test on java worker as well?

Reply:
The current tests are run on the Java worker too since that is the expected query runner. Also, `IcebergDistributedSmokeTestBase` and `IcebergDistributedTestBase` has H2 as the expected query runner, so these queries are failing in these tests.

Reply:
Got it, thanks for the explanation.